Electron upgrade 11.4.7

sadick254 opened the pull request:
# TODO
- [x] Update the CI with the proper node and npm versions
- [x] Update electron version
- [x] Update electron dependencies (electron-chromedriver and electron-mksnapshot) to match the electron version
- [x] Get the app to bootstrap on the new electron version
- [x] Once the build job passes make sure the artifacts are runnable/installable
- [x] Fix any failing tests and make sure the test jobs are passing on the CI
- [ ] Test the artifacts against the regression checklist below
- [ ] Check performance issue after upgrade on Linux (Check https://github.com/atom/atom/issues/20924 and the conversation below)

# BLOCKERS AND ISSUES
- [x] Failing github package tests
- [x] Project-wide find and replace not working.
- [x] macOS: Key binding with composition characters are ignored - alt-l inserting the `-` char
---

# Regression Checklist
## General testing

- [ ] Review issues tagged with [`electron`](https://github.com/search?utf8=%E2%9C%93&q=org%3Aatom+label%3Aelectron+is%3Aopen+is%3Aissue) to see if any of them are fixed



- [x] Verify that you can collaborate with Teletype
	- [x] Host using Atom's current stable release can collaborate with guest using Atom with the new version of Electron
	- [x] Guest using Atom's current stable release can collaborate with host using Atom with the new version of Electron
	- [x] Host and guest can collaborate when both are using Atom with the new version of Electron
  <details>
  <summary>Steps</summary>
	1. Open a file
	1. Share a portal
	1. Have a guest join the portal
	1. When guest edits the text in the file, verify that the host sees those changes
	</details>

## Check for regressions experienced in previous upgrades

### Text Input/Keybindings

See how to [setup keyboard layouts](https://github.com/atom/maintainers-log/blob/master/how-tos/setup-keyboard-layouts.md).

- [ ] IME not working (https://github.com/atom/atom/issues/14911)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Install an IME layout, for example Pinyin simplified  
  1. Open Atom and use the layout you installed as the keyboard layout
  1. Example, if you picked Pinyin simplified type in `zhongwenshuru`

  **Expected:** in Pinyin simplified `中文输入` is expected

  ![gif](https://user-images.githubusercontent.com/14314532/27643849-ccf636da-5c54-11e7-9111-e7b09469cf10.gif)

  **Issues we've seen:** Only shows latin first character and not every character typed.

  ![gif](https://user-images.githubusercontent.com/14314532/27643959-0f04396e-5c55-11e7-838c-ac15ef580fbe.gif)
  </details>

- [x] macOS: Key binding with composition characters are ignored (https://github.com/atom/atom/issues/15189)
  <details>
  <summary>Repro steps</summary>

  1. Open your keymap.cson file and add the following:
      ```cson
      'atom-text-editor':
        # use hjkl; vim-like when pressing alt (word by word, paragraph by paragraph)
        # or alt (letter by letter)
        'alt-h': 'core:move-left'
        'alt-j': 'core:move-down'
        'alt-k': 'core:move-up'
        'alt-l': 'core:move-right'
      ```
  1. Save keymap.cson and reload atom.
  1. Try the new keymappings under `ABC - Extended` keyboard layout

  **Expected**: the keybindings work

  **Issues we've seen:** it types out the modified keys and ignores the key mapping
  </details>

- [x] macOS: Composition characters mess up insertion point (https://github.com/atom/atom/issues/15344)
  <details>
  <summary>Repro steps</summary>

  1. Open keymap.cson and add the following:
      ```cson
      'atom-text-editor':
        'alt-n': 'core:move-down'
        'alt-o': 'core:move-down'
      ```
  1. Have the following in your buffer:
      ```
      Foo
      Bar
      ```
  1. Have the cursor at the beginning of file(before F character) Type alt-n, then arrow right, then insert a character. Observe that the character is inserted at the beginning of the line (where the alt-n left you).
  1. Repeat steps using <kbd>alt-o</kbd> instead of <kbd>alt-n</kbd>.

  **Expected**: Both alt-o and alt-n behave the same

  **Issues we've seen:**
  ![bug](https://user-images.githubusercontent.com/5624870/29386722-ebe4c2f6-8291-11e7-88e7-12cc89376cc7.gif)

- [ ] Ubuntu: Keystrokes involving <kbd>ctrl</kbd> resolve to the default layout instead of the active layout (https://github.com/atom/atom/issues/13170)

  <details>
  <summary>Repro steps</summary>

  **This should be tested on Linux with gnome**

  1. Install a Linux ditro with gnome, for example Ubuntu
  1. Install French AZERTY and English US QWERTY layouts
  1. Change the default layout to be French AZERTY
  1. Open Atom and switch to English US QWERTY layout
  1. Press ctrl-w

  **Expected**: `core:close` to dispatch (Or the keybinding-resolver to resolve to ctrl-w). Should be resolving to the keyboard layout that is chosen and not the OS default layout.

  **Issues we've seen:** `core:undo` dispatched because it was resolved as ctrl-z because AZERTY has Z where W is on QWERTY.
  </details>

- [x] Other keyboard layouts on new Electron version
  - [x] macOS: ABC - Extended
  - [x] macOS: Slovak
  - [x] macOS: U.S. International
  - [x] macOS: Squirrel (https://github.com/rime/squirrel)
  - [x] macOS: Full-Width Alphanumeric (Google) - Japanese layout  (https://www.google.co.jp/ime/)
  - [x] Windows: U.S.
  - [x] Linux: Pinyin Simplified

### UI

- [x] tree-view drag image (https://github.com/atom/tree-view/pull/1054)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a project in Atom with multiple files and a few folders
  1. Drag a file into a folder

  **Expected:**  
  ![image](https://cloud.githubusercontent.com/assets/482957/24454969/3b9e370e-148e-11e7-8757-53171cd6d8e3.png)

  **Issues we've seen:**

  ![image](https://user-images.githubusercontent.com/5624870/30718615-465457ce-9ed5-11e7-96e2-b367fc4d0589.gif)
  </details>

- [x] drag-and-drop indicator (https://github.com/atom/tree-view/pull/1055, https://github.com/atom/tabs/pull/426, https://github.com/atom/tabs/pull/437)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open Atom with a few folder projects on tree-view (you can do this on terminal in a directory that has existing folder and having a space between each folder. Example: `atom folderA folderB folderC`)
  1. Drag and drop all of the folders to reorder them on Atom

  **Expected:**
  The placeholder indicates where the folder/project is being dropped to.

  ![screen shot 2017-03-29 at 16 31 16](https://cloud.githubusercontent.com/assets/482957/24459995/3e78d498-149d-11e7-8037-dab5a8312822.png)

  **Issues we've seen:**
  No placeholder shows up after drag and dropping

  ![screen shot 2017-03-29 at 16 30 51](https://cloud.githubusercontent.com/assets/482957/24459997/3e9f6b1c-149d-11e7-97c5-683aaa18221a.png)
  </details>

- [x] Large file rendering (https://github.com/atom/atom/issues/16591)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a file that has over one million lines
  1. Scroll down to the bottom

  **Expected:** To not regress the number of lines that can be rendered. Rendering to be correct for lines past a certain point. Atom 1.25  can render around 800k to 1 million lines correctly.

  **Issues we've seen:** Increased number of lines rendered but bad rendering past a certain point

  ![bad renderer](https://user-images.githubusercontent.com/1058982/35113772-f2ffceda-fc82-11e7-81b6-6419c42ef3b4.gif)
  </details>

- [x] Loss of subpixel AA when the cursor is at the end of long lines (https://github.com/atom/atom/issues/16889, https://github.com/atom/atom/pull/16595)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open [text-editor.js](https://github.com/atom/atom/blob/2e19cf4a7c0a4d624b1f9c1072cd292de112cdb0/src/text-editor.js) in Atom
  1. Copy the first 21 lines to a new untitled window
  1. Turn on soft wrapping and change language mode to javascript
  1. Resize the Window so it's soft wrapped
  1. On line 15 type `a` until the you reach the end of the window

  **Expected:** To not lose subpixel AA

  **Issues we've seen:** Loss of subpixel AA. Both when soft wrapping was [enabled](https://github.com/atom/atom/pull/16943) and [disabled](https://github.com/atom/atom/pull/16595).

  ![subpixel aa loss softwrap](https://user-images.githubusercontent.com/1058982/37170628-a3901da0-230b-11e8-991e-c302db473eb3.gif)
  </details>

- [x] Scrolling horizontally shift + scroll wheel (https://github.com/atom/atom/pull/12696#issuecomment-288615912)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a file that has long lines
  1. Turn off soft wrapping and resize the window so you have a horizontal scrollbar
  1. Press shift and scroll in both directions using the mouse wheel

  **Expected:** File to scroll horizontally in both directions

  **Issues we've seen:** File does not scroll horizontally at all in any direction
  </details>

- [x] Scrollbars misbehaving on the first file that is opened (https://github.com/atom/atom/pull/12696#issuecomment-251299649)
  <details>
  <summary>Repro steps</summary>

  1. Open Atom
  1. Open some file that has content enough to have a vertical scrollbar
  1. Close Atom
  1. Open Atom and look at the vertical scrollbar

  **Expected:** Scrollbar to be visible

  **Issues we've seen:** Scrollbar is not visible and is flickering when you are editing
  </details>

- [x] Middle clicking on unsaved tab (https://github.com/atom/atom/issues/15197)
	- [x] macOS
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open Atom and open a new unsaved file.
  1. Make edits to the unsaved file and middle click on the tab.

  **Expected:**
  Clicking save/cancel or the options on the dialog works.

  **Issues we've seen:**
  The UI and the dialog is unable to receive mouse clicks. You can still choose options via Keyboard, but not mouse.
  </details>

- [x] Linux: Atom scrolls even when not focused (https://github.com/atom/atom/issues/15482)

  <details>
  <summary>Repro steps</summary>

  1. Open Atom with a large enough file that has a scrollbar
  1. Open a different application and place it over the Atom window
  1. Scroll with the mouse inside the other window
  1. Focus the Atom window

  **Expected:** Atom window should keep the original scroll position

  **Issues we've seen:** The Atom window scrolls after it is focused
  Note: this is a regression that has not been fixed since Electron 3, will leave it in the template to keep it on our radar once it's fixed 
  </details>

### Other

- [x] Supported Versions of OS (https://github.com/atom/atom/issues/15297)
	- [x] macOS
	- [ ] Windows 10
	- [ ] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Check the list of supported platforms [Electron Master](https://github.com/electron/electron/blob/master/docs/tutorial/support.md#supported-platforms)
  1. Check what version is listed as minimum for each platform on https://atom.io and https://atom.io/faq

  **Expected:** https://atom.io and https://atom.io/faq to display the same versions as the Electron documentation

  **Issues we've seen:** macOS version support changed and https://atom.io and https://atom.io/faq was not updated
  </details>

- [x] macOS : Native tabs (https://github.com/atom/atom/pull/14711)
  <details>
  <summary>Repro steps</summary>

  1. Open two atom windows on mac
  1. Go to View in menus and check `Show Tab Bar`
  ![screen shot 2017-06-03 at 5 10 32 pm](https://cloud.githubusercontent.com/assets/378023/26751932/aedfc05a-487f-11e7-9643-8047242b3bcb.png)
  1. Do this for both Atom windows.
  1. Drag one atom window to the other via tab bar

  **Expected:**
  Native Tab bar shows up and able to drag different windows to each other.  Also make sure UI isn't messed up.

  **Issues we've seen:**
  Native Tab Bar doesn't work and messes up UI when enabled
  </details>

- [ ] Ubuntu with KDE: Menu uses 100% CPU (https://github.com/electron/electron/issues/8455, https://github.com/atom/atom/issues/13885)
  <details>
  <summary>Repro steps</summary>

  1. Install Atom on Ubuntu KDE. [[instructions](https://linuxconfig.org/how-to-install-kde-plasma-desktop-on-ubuntu-18-04-bionic-beaver-linux)]
  1. Enable KDE plasma global menu: Right-click on the desktop -> Add Panel => Application Menu Bar. [[demo](https://youtu.be/qo5fhiXAUbI)]
  1. Start Atom.
  1. Try to use the desktop menu.

  **Expected:**
  Desktop menu items work

  **Issues we've seen:**
  Desktop menu items do nothing when you click on things.
  </details>

- [x] macOS: Slovak QWERTZ (https://github.com/atom/atom-keymap/issues/223)
  <details>
  <summary>Repro steps</summary>

  1. Install Slovak keyboard layout on mac
  1. Open keyboard viewer to see the keys are are suppose to show up and open keyboard resolver on atom.
  1. hit `cmd+'`.

  **Expected:**  Resolves to `cmd+'` like it shows on keyboard viewer on mac.

  **Issues we've seen:** It resolves to `ctrl+§`
  </details>

- [x] macOS: IME jump (https://github.com/atom/atom/issues/15696)
  <details>
  <summary>Repro steps</summary>

  1. Install an IME layout, for example Chinese pinyin
  1. Type one character

  **Expected:**
  Expected the IME to be positioned at under the character you entered
  **Issues we've seen:** The IME window is in the top left corner. When you enter the second character it jumps to be positioned under the text
  </details>

DeeDeeG commented:
Taking a look at the CI failures, it looks like there are some build errors for a couple packages (on macOS and 32-bit Windows. Windows 64-bit and Linux finished building.)

#### macOS

Package `@atom/fuzzy-native`

<details>

```
Installing locked modules ✗
> atom@1.60.0-dev preinstall /Users/runner/work/1/s
> node -e 'process.exit(0)'


> @atom/fuzzy-native@1.2.0 install /Users/runner/work/1/s/node_modules/@atom/fuzzy-native
> node-gyp rebuild

  CXX(target) Release/obj.target/fuzzy-native/src/binding.o
  CXX(target) Release/obj.target/fuzzy-native/src/score_match.o
  CXX(target) Release/obj.target/fuzzy-native/src/MatcherBase.o

In file included from ../src/binding.cpp:1:
In file included from ../../../nan/nan.h:54:
In file included from /Users/runner/.atom/.node-gyp/Library/Caches/node-gyp/11.4.7/include/node/node.h:67:
In file included from /Users/runner/.atom/.node-gyp/Library/Caches/node-gyp/11.4.7/include/node/v8.h:30:
/Users/runner/.atom/.node-gyp/Library/Caches/node-gyp/11.4.7/include/node/v8-internal.h:418:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:697:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
In file included from ../src/binding.cpp:1:
In file included from ../../../nan/nan.h:2884:
                                                 ^
In file included from ../src/binding.cpp:1:
In file included from ../../../nan/nan.h:2884:
../../../nan/nan_typedarray_contents.h:34:43: warning: 'GetContents' is deprecated: Use GetBackingStore. See http://crbug.com/v8/9908. [-Wdeprecated-declarations]
      data   = static_cast<char*>(buffer->GetContents().Data()) + byte_offset;
                                          ^
/Users/runner/.atom/.node-gyp/Library/Caches/node-gyp/11.4.7/include/node/v8.h:5349:3: note: 'GetContents' has been explicitly marked deprecated here
make: *** Waiting for unfinished jobs....
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/runner/work/1/s/apm/node_modules/atom-package-manager/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:223:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/Users/runner/work/1/s/apm/node_modules/atom-package-manager/bin/node" "/Users/runner/work/1/s/apm/node_modules/atom-package-manager/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/runner/work/1/s/node_modules/@atom/fuzzy-native
gyp ERR! node -v v12.14.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @atom/fuzzy-native@1.2.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @atom/fuzzy-native@1.2.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/runner/.atom/.apm/_logs/2021-07-02T20_30_47_162Z-debug.log
child_process.js:651
    throw err;
    ^

Error: Command failed: /Users/runner/work/1/s/apm/node_modules/atom-package-manager/bin/apm ci
    at checkExecSyncError (child_process.js:630:11)
    at Object.execFileSync (child_process.js:648:15)
    at module.exports (/Users/runner/work/1/s/script/lib/run-apm-install.js:12:16)
    at Object.<anonymous> (/Users/runner/work/1/s/script/bootstrap:47:1)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 49152,
  stdout: null,
  stderr: null
}
```

</details>

#### Windows x86

Package `pathwatcher`

<details>

```
Installing locked modules failed
> atom@1.60.0-dev preinstall D:\a\1\s
> node -e 'process.exit(0)'


> @atom/fuzzy-native@1.2.0 install D:\a\1\s\node_modules\@atom\fuzzy-native
> node-gyp rebuild


D:\a\1\s\node_modules\@atom\fuzzy-native>if not defined npm_config_node_gyp (node "D:\a\1\s\apm\node_modules\atom-package-manager\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "D:\a\1\s\apm\node_modules\atom-package-manager\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild ) 
  binding.cpp
  score_match.cpp
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(21): warning C4305: 'initializing': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(24): warning C4305: 'initializing': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(27): warning C4305: 'initializing': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(108): warning C4305: '=': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(111): warning C4305: '=': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(113): warning C4305: '=': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(115): warning C4305: '=': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(134): warning C4305: '*=': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\score_match.cpp(151): warning C4305: '=': truncation from 'double' to 'float' [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
  MatcherBase.cpp
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): warning C4309: 'static_cast': truncation of constant value (compiling source file ..\src\binding.cpp) [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(9182): warning C4996: 'v8::MicrotasksCompletedCallback': Use *WithData version. (compiling source file ..\src\binding.cpp) [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
  c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(7384): note: see declaration of 'v8::MicrotasksCompletedCallback' (compiling source file ..\src\binding.cpp)
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(9190): warning C4996: 'v8::MicrotasksCompletedCallback': Use *WithData version. (compiling source file ..\src\binding.cpp) [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
  c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(7384): note: see declaration of 'v8::MicrotasksCompletedCallback' (compiling source file ..\src\binding.cpp)
d:\a\1\s\node_modules\@atom\fuzzy-native\src\matcherbase.cpp(26): warning C4293: '<<': shift count negative or too big, undefined behavior [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
d:\a\1\s\node_modules\@atom\fuzzy-native\src\matcherbase.cpp(28): warning C4334: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?) [D:\a\1\s\node_modules\@atom\fuzzy-native\build\fuzzy-native.vcxproj]
  win_delay_load_hook.cc
     Creating library D:\a\1\s\node_modules\@atom\fuzzy-native\build\Release\fuzzy-native.lib and object D:\a\1\s\node_modules\@atom\fuzzy-native\build\Release\fuzzy-native.exp
  Generating code
  All 1241 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  fuzzy-native.vcxproj -> D:\a\1\s\node_modules\@atom\fuzzy-native\build\Release\\fuzzy-native.node

> vscode-ripgrep@1.9.0 postinstall D:\a\1\s\node_modules\vscode-ripgrep
> node ./lib/postinstall.js

Finding release for v12.1.1
GET https://api.github.com/repos/microsoft/ripgrep-prebuilt/releases/tags/v12.1.1
Downloading from https://api.github.com/repos/microsoft/ripgrep-prebuilt/releases/assets/22625165
Downloading to C:\Users\VSSADM~1\AppData\Local\Temp\vscode-ripgrep-cache-1.9.0\ripgrep-v12.1.1-i686-pc-windows-msvc.zip
Downloading with Invoke-WebRequest
Unzipping to D:\a\1\s\node_modules\vscode-ripgrep\bin
Expand-Archive completed

> pathwatcher@8.1.1 install D:\a\1\s\node_modules\pathwatcher
> node-gyp rebuild


D:\a\1\s\node_modules\pathwatcher>if not defined npm_config_node_gyp (node "D:\a\1\s\apm\node_modules\atom-package-manager\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "D:\a\1\s\apm\node_modules\atom-package-manager\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild ) 
  main.cc
  common.cc
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): error C2220: warning treated as error - no 'object' file generated (compiling source file ..\src\common.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): warning C4309: 'static_cast': truncation of constant value (compiling source file ..\src\common.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): error C2220: warning treated as error - no 'object' file generated (compiling source file ..\src\main.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): warning C4309: 'static_cast': truncation of constant value (compiling source file ..\src\main.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
  handle_map.cc
  pathwatcher_win.cc
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): error C2220: warning treated as error - no 'object' file generated (compiling source file ..\src\handle_map.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): warning C4309: 'static_cast': truncation of constant value (compiling source file ..\src\handle_map.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): error C2220: warning treated as error - no 'object' file generated (compiling source file ..\src\pathwatcher_win.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): warning C4309: 'static_cast': truncation of constant value (compiling source file ..\src\pathwatcher_win.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
  win_delay_load_hook.cc

gyp ERR! build error 
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (D:\a\1\s\apm\node_modules\atom-package-manager\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:223:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "D:\\a\\1\\s\\apm\\node_modules\\atom-package-manager\\bin\\node.exe" "D:\\a\\1\\s\\apm\\node_modules\\atom-package-manager\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd D:\a\1\s\node_modules\pathwatcher
gyp ERR! node -v v12.14.1
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pathwatcher@8.1.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pathwatcher@8.1.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\VssAdministrator\.atom\.apm\_logs\2021-07-02T20_30_15_383Z-debug.log
child_process.js:651
    throw err;
    ^

Error: Command failed: D:\a\1\s\apm\node_modules\atom-package-manager\bin\apm.cmd ci
    at checkExecSyncError (child_process.js:630:11)
    at Object.execFileSync (child_process.js:648:15)
    at module.exports (D:\a\1\s\script\lib\run-apm-install.js:12:16)
    at Object.<anonymous> (D:\a\1\s\script\bootstrap:47:1)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 5128,
  stdout: null,
  stderr: null
}
```

</details>

(If I had to guess, I don't think the package and renderer tests would pass if the builds completed so the tests could run)

sadick254 commented:
I don't think the tests would pass either. We first need to get the build passing and start working on the failing tests.

DeeDeeG commented:
I have a CI run with all of those packages mentioned above updated, to fix the build errors.

It builds Atom on all but Windows 32-bit. https://dev.azure.com/DeeDeeG/b/_build/results?buildId=1180

_(macOS hits a test error very early, in the core main-process tests during the build job, so it did not publish an artifact and the "build" job reports as failed, but it did actually finish building.)_

Here are the issues I'm seeing:

#### Windows 32-bit build errors

Windows 32-bit has errors building the `pathwatcher` package; There are source C++ files that have compiler warnings, and apparently a flag is set to error out if there are warnings. So the C++ files get no output built for them, and the compilation of the package errors out in general... 

Here is the compiler error documentation ([C2220](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2220?view=msvc-160))

Example error from the build:
```
2021-07-06T01:15:20.9744079Z c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): error C2220: warning treated as error - no 'object' file generated (compiling source file ..\src\common.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
```

Example warning ([C4309](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-2-c4309?view=msvc-160)) as the cause of the above error:

_(This warning is about some constants' types being too small for the data in them, leading to over-sized data being chopped smaller?? I think we need to use a bigger constant type?)_
```
2021-07-06T01:15:20.9742630Z c:\users\vssadministrator\appdata\local\node-gyp\cache\11.4.7\include\node\v8.h(5441): warning C4309: 'static_cast': truncation of constant value (compiling source file ..\src\main.cc) [D:\a\1\s\node_modules\pathwatcher\build\pathwatcher.vcxproj]
```

**Solution?**: I suggest someone with C++ knowledge can go in and fix the compiler warnings, or we can find a way to turn off the "treat warnings as errors" flag for the compiler.

#### Other platforms/OSes (Web Components v0 is deprecated, move to Custom Elements v1)

Other platforms/OSes run into the `document.registerElement is not a function` problem as mentioned and discussed here: https://github.com/atom/atom/pull/22011#issuecomment-798965159 and here: https://github.com/atom/atom/pull/22125

**Solution?**: Use a Web Components v0 polyfill, or dig into the Custom Elements v1 specification, and learn to port the custom elements we use to the v1 specification. _(See work-in-progress attempts linked to just above.)_

DeeDeeG commented:
More Windows 32-bit problems.

I opened a PR for this one: https://github.com/atom/atom-keymap/pull/250

It is a similar issue with [`keyboard-layout`](https://github.com/atom/keyboard-layout), although this was already fixed in https://github.com/atom/keyboard-layout/pull/59 and released in `keyboard-layout@2.0.17`. But the new `keyboard-layout` package version still needs to be bumped to at least 2.0.17 in `atom-keymap`'s [`package.json`](https://github.com/atom/atom-keymap/blob/v8.2.14/package.json#L30). So that's what my PR does.

sadick254 commented:
@DeeDeeG I will be using your tree-sitter fork to get the macOS build passing for now.

DeeDeeG commented:
Okay.

I think for best security, I would recommend putting the exact commit hash in the URL in `package.json`, rather than the GitHub branch name. Because the branch can be updated on GitHub, whereas at least forcing a repeated commit hash in git is super hard/unlikely.

So I suppose it would be :
```json
"tree-sitter": "https://github.com/DeeDeeG/node-tree-sitter#bb298eaae66e0c4f11908cb6209f3e141884e88e"
```

(One can do `npm install --package-lock-only https://github.com/DeeDeeG/node-tree-sitter#bb298eaae66e0c4f11908cb6209f3e141884e88e`)

That is my commit https://github.com/DeeDeeG/node-tree-sitter/commit/bb298eaae66e0c4f11908cb6209f3e141884e88e bumping `tree-sitter@0.17.1` to compile as  C++14. That (0.17.1) is the version of `tree-sitter` Atom is currently using: https://github.com/atom/atom/blob/v1.58.0-beta0/package.json#L164

_(Thoughts for the future: Eventually, perhaps the Atom github org should fork the tree-sittter/node-tree-sitter repo and publish it to the npm registry as `@atom/tree-sitter`? Or request the `tree-sitter` org members publish another release in the 0.17.x series? Or hope they merge https://github.com/tree-sitter/node-tree-sitter/pull/83, and upgrade Atom to `tree-sitter@0.19`/newer.)_

mfonville commented:
@sadick254 how about going to `v11.4.10` of Electron? It has some security fixes and should have same compatibility

mfonville commented:
Electron 11.4.11 is now available https://github.com/atom/atom/pull/22810

sadick254 commented:
Thank you @mfonville for following up on the Electron upgrade and Electron releases. We would update to the latest 11.x.x after getting this working. We did start with using `customElements` instead of the `deprecated document.registerElement`

sadick254 committed:
Bump packages

- @atom/fuzzy-native@1.2.1
- @atom/nsfw@1.0.28",
- atom-keymap@8.2.15",
- first-mate@7.4.3",
- pathwatcher@8.1.2",

sadick254 committed:
Bump electron version

sadick254 committed:
Stub global.atom

The test don't require the atom environment to be setup.

sadick254 committed:
Bump github@0.36.10

sadick254 committed:
Add timeout time

DeeDeeG commented:
I see this passes CI now. Very exciting! Thank you for all the work on this.

I have played around with some very basic usage on Windows and Linux, and it's working well so far! I didn't test many things, but the things I tried seem to be working. :+1:  :+1:  :+1:  :+1:  :+1:  :+1:

sadick254 commented:
This upgrade will break packages that are still using `document.registerElement `in their implementations. i.e . [busy-signal](https://github.com/steelbrain/busy-signal) and others.

sadick254 commented:
It is probably better to add some sort of polyfill for `document.registerElement`. This will make sure packages that use `document.registerElement` will continue to work. The last thing we want to do as Atom maintainers is introduce a change that has the potential of breaking many packages.

sadick254 committed:
Add document.registerElement polyfill

Atom is upgrading to use electron 11.4.7 however on that version
chromium has deprecated `document.registerElement`.

There are some community packages whose implementations depend on
`document.registerElement`. Therefore shipping without a polyfill will
break those packages.

As Atom maintainers we wouldn't want to introduce a change
that has the potential of breaking many packages.

sadick254 committed:
Add document.registerElement polyfill (#22901)

Atom is upgrading to use electron 11.4.7 however on that version
chromium has deprecated `document.registerElement`.

There are some community packages whose implementations depend on
`document.registerElement`. Therefore shipping without a polyfill will
break those packages.

As Atom maintainers we wouldn't want to introduce a change
that has the potential of breaking many packages.

DeeDeeG commented:
FYI/heads up:

Some packages have issues. I tried `ide-python` and `ide-javascript` and there is an issue with [an assert](https://github.com/atom-community/atom-ide-base/blob/v3.4.0/nuclide/nuclide-commons-ui/Dropdown.js#L26). Perhaps it has to do with the `remote` module being deprecated and off by default.

I tried to install `hydrogen`, but it had some error. I didn't have time to look into that yet.

The usual reliable `terminus` works like a charm, at least on Windows. :+1:

mfonville committed:
Upgrade to electron 11.4.12 for security fixes (#22810)

* Upgrade to electron 11.4.11 for security fixes

* Electron 11.4.12

sadick254 commented:
@DeeDeeG We have enabled remote module https://github.com/atom/atom/blob/10f7d454edb3d61e3c85f6be1c58122b656c295f/src/main-process/atom-window.js#L61

The remote module has not been removed on electron version 11. It will be removed on version 12. I think the issue might be something else.

mfonville commented:
Electron now went to 11.5.0 https://github.com/electron/electron/releases/tag/v11.5.0 with some security fixes

It will be the final release for the 11-series.

the-j0k3r commented:
Holy mother of electrons, wut?

Congrats, keep it going

darangi commented:
@DeeDeeG Thanks for the heads up 🙇🏾 , I went ahead to install about 41 ide packages(The most I could find), I can confirm only `ide-javascript` and `ide-python` throw those errors. Also, Installing hydrogen does not work, it is stuck at installing. 

I will create issues in the repos responsible for those packages

UziTech commented:
@darangi @sadick254 what is needed to get this over the line?

sadick254 commented:
I thought ` ide-javascript` and `ide-python` that had errors would be fixed. But that doesn't seem to be the case. I see the errors from ` ide-javascript` and `ide-python` as non-blockers, those packages are external.

the-j0k3r commented:
Congrats. @sadick254 great job on this.

DeeDeeG commented:
I'm glad to see this merged!!!

One small request: Would the Atom GitHub org be willing to fork `tree-sitter`? That way I don't have to keep my fork live just to ensure Atom's `master` branch can build... I will keep my fork up as long as needed, obviously... I love Atom and want it to be able to build. but I think it's good to not have my personal GitHub fork of `tree-sitter` still "in the loop" for building official Atom. Thank you!

<details>

Atom's `master` branch as of this PR needs my fork of `tree-sitter` to be available on GitHub in order to build. See: https://github.com/atom/atom/blob/7033a5a22f260a54c0ffb69f45d12622478778df/package.json#L164

Can an official maintainer of Atom, or the `atom` GitHub org, host the forked `tree-sitter` so I don't have to be "in the loop", and official Atom's `master` branch can build without my fork?

Example workflow to do this:

- Fork https://github.com/tree-sitter/node-tree-sitter under the `atom` GitHub org, using github.com
- Clone it locally `git clone https://github.com/atom/node-tree-sitter` AND `cd node-tree-sitter`
- Add my fork as a remote `git remote add deedeeg https://github.com/DeeDeeG/node-tree-sitter` AND `git remote update`
- Check out the commit with my patch as a new branch `git switch -c [arbitrary-branch-name-here, for example "0.17.1-C++-14"] bb298eaae66e0c4f11908cb6209f3e141884e88e`
  - (OR just manually recreate my patch https://github.com/DeeDeeG/node-tree-sitter/commit/bb298eaae66e0c4f11908cb6209f3e141884e88e and commit that on some branch...)
- Push the branch with the patch to the `atom` GitHub org's fork... `git push --set-upstream origin [whatever-branch-name-has-the-patch, for example "0.17.1-C++-14"]`
- Then you can switch the `tree-sitter` dependency in Atom's `package.json` to the following...
  -  `"tree-sitter": "git+https://github.com/atom/node-tree-sitter.git#bb298eaae66e0c4f11908cb6209f3e141884e88e",`
    - Or if you manually recreated the patch, then specify whatever your actual resulting commit hash is, instead of `bb298ea`...

</details>

baarkerlounger commented:
Will there be a new beta now that this has been merged or are there other things that need to be done first?

iamunadike commented:
Please @sadick254  I understand aabout the archiving of at atom by microsoft, but I am among those that I already love the editor so much to give up on it anytime soon, most people say this and that about VScode and the likes but forget atom is the most hackable editor till date, 
Infact it too precise,
With that in mind , I will like to appeal for your continuous work as a great team of developers are willing to work with you using atom fork to continue the development and releases. God bless you Sir for all the effort and work you have put in place to keep this atom project alive and others too who contributed
PLEASE lets not give up, we Have come this far. we shall PREVAIL

ZodiacYouth commented:
HEHEHE HOW MUCH???

On Tue, 28 Jun 2022, 16:54 Michael You, ***@***.***> wrote:

> Please @sadick254 <https://github.com/sadick254> I understand aabout the
> archiving of at atom by microsoft, but I am among those that I already love
> the editor so much to give up on it anytime soon, most people say this and
> that about VScode and the likes but forget atom is the most hackable editor
> till date,
> Infact it too precise,
> With that in mind , I will like to appeal for your continuous work as a
> great team of developers are willing to work with you using atom fork to
> continue the development and releases. God bless you Sir for all the effort
> and work you have put in place to keep this atom project alive and others
> too who contributed
> PLEASE lets not give up, we Have come this far. we shall PREVAIL
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/atom/atom/pull/22687#issuecomment-1168832282>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN6ZJYAFTN5ZKC4U2TXZIV3VRMG27ANCNFSM47XH6VQA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
